src/App.tsx

```diff
@@ -15,47 +16,130 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((a, b) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return a.localeCompare(b);
+      case SortType.LENGTH:
+        return a.length - b.length;
+      default:
+        return SortType.NONE;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export const App: React.FC<{}> = () => {
+  const [firstState, secondState] = useState<State>({
+    isReversed: false,
+    sortType: SortType.NONE,
+  });
+
+  const sortAlphabet = () => {
+    secondState((prev) => ({
+      sortType: SortType.ALPHABET,
+      isReversed: prev.isReversed,
+    }));
+  };
+
+  const sortLength = () => {
+    secondState((prev) => ({
+      sortType: SortType.LENGTH,
+      isReversed: prev.isReversed,
+    }));
+  };
```

Comment:
you can join these function, because they do the same. Just pass a sortType as an argument

Reply:
sure!